packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -181,39 +181,102 @@ const AutoBullet: BuildInEditFeature<PluginKeyboardEvent> = {
             () => {
                 let regions: RegionBase[];
                 let searcher = editor.getContentSearcherOfCursor();
-                let textBeforeCursor = searcher.getSubStringBefore(5);
+                let textBeforeCursor = searcher.getSubStringBefore(4);
                 let textRange = searcher.getRangeFromText(textBeforeCursor, true /*exactMatch*/);
-                let listStyle;
-                let listType;
 
-                if (editor.isFeatureEnabled(ExperimentalFeatures.AutoFormatList)) {
-                    const listInfo = getListInfo(textBeforeCursor);
-                    listType = listInfo.listType;
-                    listStyle = listInfo.listStyle;
-                } else {
-                    listType =
-                        textBeforeCursor.indexOf('*') == 0 || textBeforeCursor.indexOf('-') == 0
-                            ? ListType.Unordered
-                            : isAListPattern(textBeforeCursor)
-                            ? ListType.Ordered
-                            : ListType.None;
-                }
                 if (!textRange) {
                     // no op if the range can't be found
-                } else if (listType === ListType.Unordered) {
+                } else if (
+                    textBeforeCursor.indexOf('*') == 0 ||
+                    textBeforeCursor.indexOf('-') == 0
+                ) {
                     prepareAutoBullet(editor, textRange);
-                    toggleBullet(editor, listStyle as BulletListType);
-                } else if (listType === ListType.Ordered) {
+                    toggleBullet(editor);
+                } else if (isAListPattern(textBeforeCursor)) {
                     prepareAutoBullet(editor, textRange);
-                    toggleNumbering(
-                        editor,
-                        undefined /* startNumber*/,
-                        listStyle as NumberingListType | undefined
-                    );
+                    toggleNumbering(editor);
                 } else if ((regions = editor.getSelectedRegions()) && regions.length == 1) {
                     const num = parseInt(textBeforeCursor);
                     prepareAutoBullet(editor, textRange);
-                    toggleNumbering(editor, num, listStyle as NumberingListType);
+                    toggleNumbering(editor, num);
+                }
+                searcher.getRangeFromText(textBeforeCursor, true /*exactMatch*/)?.deleteContents();
+            },
+            null /*changeSource*/,
+            true /*canUndoByBackspace*/
+        );
+    },
+};
+
+/**
+ * AutoBulletList edit feature, provides the ability to automatically convert current line into a bullet list.
+ */
+const AutoBulletList: BuildInEditFeature<PluginKeyboardEvent> = {
```

Comment:
Besides we can disable a feature separately, is there any other reason we want to split into two features?